src/nvim/autocmd.c

```diff
@@ -2013,6 +2013,7 @@ char_u *getnextac(int c, void *cookie, int indent, bool do_concat)
   }
 
   AutoCmd *ac = acp->nextcmd;
+  bool oneshot = ac->once;
```

Comment:
@lewis6991 If an autocmd is `++once` and then returns false `oneshot` is now false and it won't be removed. This is what [I meant](https://github.com/neovim/neovim/pull/17784/files#r830526328)

Reply:
I believe that is not the intention.

Reply:
I just think that if the autocmd was defined as once, it should run just once. Make the user pick ++once or just detach when they return true but not a mix of both.

Reply:
I agree and believe that is the intention of this PR.

From what I can tell `once` cannot be set to `false`. It's either set via `++once` or set via the callback return.

Reply:
Yes @lewis6991 is right, if the autocommand returns false the value of oneshot is not changed. So if it’s a `once` autocommand it is still removed.

Reply:
Oh now I get it 😄